src/test/credentials/diskCache.test.ts

```diff
@@ -0,0 +1,187 @@
+    const startUrl = 'https://123456.awsapps.com/start'
+    let registrationFilename: string
+    let accessTokenFileName: string
+    const HOUR_IN_MS = 3600000
+    const FOURTEEN_MINS_IN_MS = 840000
+    let sut: DiskCache
+
+    const validRegistration: SsoClientRegistration = {
+        clientId: 'dummyId',
+        clientSecret: 'dummySecret',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    const validAccessToken: SsoAccessToken = {
+        startUrl: startUrl,
+        region: ssoRegion,
+        accessToken: 'longstringofrandomcharacters',
+        expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+    }
+
+    beforeEach(async () => {
+        // Make a temp folder for all these tests
+        tempFolder = await makeTemporaryToolkitFolder()
+        registrationFilename = path.join(tempFolder, `aws-toolkit-vscode-client-id-${ssoRegion}.json`)
+        accessTokenFileName = path.join(tempFolder, 'c1ac99f782ad92755c6de8647b510ec247330ad1.json')
+        sut = new DiskCache(tempFolder)
+    })
+
+    afterEach(async () => {
+        await rmrf(tempFolder)
+    })
+
+    describe('loadClientRegistration', () => {
+        it('should return a valid registration', () => {
+            writeFileSync(registrationFilename, JSON.stringify(validRegistration))
+
+            const returnedRegistration = sut.loadClientRegistration(ssoRegion)
+
+            assert.deepStrictEqual(returnedRegistration, validRegistration)
+        })
+
+        it('should return undefined if no registration exists', () => {
+            const returnedRegistration = sut.loadClientRegistration(ssoRegion)
+            assert.strictEqual(returnedRegistration, undefined)
+        })
+
+        it('should return undefined for expired registration', () => {
+            const expiredRegistration: SsoClientRegistration = {
+                clientId: 'dummyId',
+                clientSecret: 'dummySecret',
+                expiresAt: new Date(Date.now() - HOUR_IN_MS).toISOString(),
+            }
+
+            writeFileSync(registrationFilename, JSON.stringify(expiredRegistration))
+
+            const returnedRegistration = sut.loadClientRegistration(ssoRegion)
+            assert.strictEqual(returnedRegistration, undefined)
+        })
+
+        it('should return undefined if within 15 minutes from expiration', () => {
+            const expiredSoonRegistration: SsoClientRegistration = {
+                clientId: 'dummyId',
+                clientSecret: 'dummySecret',
+                expiresAt: new Date(Date.now() + FOURTEEN_MINS_IN_MS).toISOString(),
+            }
+
+            writeFileSync(registrationFilename, JSON.stringify(expiredSoonRegistration))
+
+            const returnedRegistration = sut.loadClientRegistration(ssoRegion)
+            assert.strictEqual(returnedRegistration, undefined)
+        })
+    })
+    describe('saveClientRegistration', () => {
+        if (process.platform !== 'win32') {
```

Comment:
Equivalent test for Windows?

Reply:
From the Node.js docs: "Caveats: on Windows only the write permission can be changed, and the distinction among the permissions of group, owner or others is not implemented."  Based on my testing it defaults to 644.  I suppose I could write a test that at least tests for the first digit to be a '6'.